stablehlo/dialect/StablehloOps.cpp

```diff
@@ -4797,7 +4797,7 @@ ParseResult parseConvolutionDimensions(AsmParser& parser,
     }
 
     llvm::SmallDenseMap<int64_t, int64_t> spatialDimsMap;
-    constexpr int64_t kInvalidDimension = -1;
```

Comment:
As discussed in the team meeting, can you also check for `< 0`. Also, we'll sync separately on what to do with MLIR files. It appears that the following two changes will be sufficient: 1) prettyprinter for TypeExtensions, 2) shape inference testing infra to prettyprint -1 as ?.